pkg/runtime/reconciler.go

```diff
@@ -342,8 +342,8 @@ func (r *resourceReconciler) patchResourceMetadataAndSpec(
 	rlog.Enter("kc.Patch (metadata + spec)")
 	err = r.kc.Patch(
 		ctx,
-		latest.RuntimeObject(),
-		client.MergeFrom(desired.RuntimeObject()),
+		latest.RuntimeObject().DeepCopyObject(),
```

Comment:
I intentionally did not use the DeepCopyObject() here and I hope this does not introduce the exception that underlying object has been changed.

The doc of this method says that pass the pointer of the object so that changes applied from patch can be reflected back. If we use .DeepCopyObject() then latest will not have the changes reflected back in latest object.

if this new way works, I may have misunderstood the working. Can you share your delve logs to help my understanding?

Also you will have to update the unit tests to incorporate this new DeepCopyObject method.

Reply:
Do we really need to `DeepCopyObject()` the `latest` here?